app/controllers/groups_controller.rb

```diff
@@ -587,6 +587,20 @@ def populate_repo_with_starter_files
     redirect_to course_assignment_path(current_course, assignment)
   end
 
+  def match_students
+    grouping_ids = params[:groupings]
+    exam_template_id = params[:exam_template]
+    groupings = Grouping.find(grouping_ids)
```

Comment:
We can do a better job of checking ids here to ensure only associated objects are used.

1. pass the `assignment_id` (from the front-end) and use it to obtain the assignment
2. Use `assignment.exam_templates.find` and `assignment.groupings.find`